rust/osm2lanes/src/transform.rs

```diff
@@ -696,13 +720,45 @@ pub fn lanes_to_tags(lanes: &[LaneSpec], cfg: &Config) -> Result<Tags, LaneSpecE
     {
         let left_cycle_lane = lanes
             .iter()
-            .take_while(|lane| lane.lane_type != LaneType::Driving)
-            .find(|lane| lane.lane_type == LaneType::Biking);
+            .take_while(|lane| {
+                !matches!(
+                    lane,
+                    Lane::Travel {
+                        designated: LaneDesignated::Motor,
+                        ..
+                    }
+                )
+            })
+            .find(|lane| {
+                matches!(
+                    lane,
+                    Lane::Travel {
+                        designated: LaneDesignated::Bicycle,
+                        ..
+                    }
+                )
+            });
         let right_cycle_lane = lanes
             .iter()
             .rev()
-            .take_while(|lane| lane.lane_type != LaneType::Driving)
-            .find(|lane| lane.lane_type == LaneType::Biking);
+            .take_while(|lane| {
```

Comment:
And yeah... in terms of readability, the one-liner was so much better. :\   I might add the predicates after this PR.

Reply:
yeah, I think it can be attributed to the increased complexity in the schema?